com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Builder.java

```diff
@@ -26,6 +30,29 @@
  */
 public abstract class Builder {
 
+    private static Map<Class<?>,  byte[]> classToParamsDigest = new HashMap<Class<?>,  byte[]>();
+
+    public static void addParamsDigest(Class<?> klass, Map<String, String> options, BuilderParams builderParams) throws NoSuchAlgorithmException {
```

Comment:
I don't want to do it too often, so I do it once per Builder class when we scan for classes.